deepinv/physics/mcmri.py

```diff
@@ -0,0 +1,77 @@
+import torch
+from torch import nn
+from deepinv.physics.forward import LinearPhysics
+
+from deepinv.physics import adjoint_function
+
+
+
+
+class MultiCoilMRI(LinearPhysics):
+    r"""
+    Multi-coil MRI operator.
+
+    The linear operator is defined as:
+    .. math::
+
+        y_c = \text{diag}(p) F \text{diag}(s_c) x
+
+        for c=1,\dots,C coils, where y_c are the measurements from the cth coil, \text{diag}(p) is the acceleration mask, F is the Fourier transform and \diag(s_c) is the cth coil sensitivity.
+    
+    :param torch.Tensor mask: binary sampling mask which should have shape [B,1,H,W].
+    :param torch.Tensor coil_maps: complex valued coil sensitvity maps which should have shape [B,C,H,W].
+    :param device: specify which device you want to use (i.e, cpu or gpu).
+    """
+
+    def __init__(
+        self,
+        mask,
+        coil_maps,
+        device=torch.device("cpu"),
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.mask = mask.to(device)
+        self.coil_maps = coil_maps.to(device)
+
+    def A(self, x, **kwargs):
+        r"""  
+        Applies linear operator.  
+        
+        :param torch.Tensor x: image with shape [B,H,W,2].  
```

Comment:
`x` should be B,2,H,W to be consistent with `deepinv.physics.MRI`, same in `A_adjoint`

Reply:
Good point. Done

Reply:
@tachella will there be any problems if `x` is of shape `B,2,H,W` but `y` is of shape `B,2,N,H,W` i.e. an extra dim? Maybe in `generate_dataset` or something?